ports/stm32f4/common-hal/pulseio/PWMOut.c

```diff
@@ -68,8 +68,9 @@ STATIC uint32_t timer_get_source_freq(uint32_t tim_id) {
 }
 
 STATIC uint32_t timer_get_internal_duty(uint16_t duty, uint32_t period) {
-    //duty cycle is duty/0xFFFF fraction x (number of pulses per period)
-    return (duty*period) / ((1 << 16) - 1);
+    //duty cycle is (0xFFFF - duty)/0xFFFF fraction x (number of pulses per period)
+    //Note that pulses are inverted, so duty cycle is inverted
```

Comment:
Does this break the PWMOut API? I was thinking that Display would take a kwarg to invert brightness rather than changing PWMOut at all.

Reply:
Nope, I wasn't complete enough in my original testing of PWMOut - ST counts pulses as logic lows, so this change actually brings all of PWMOut in line with the rest of circuitpython. I was used to the open drain system and didn't notice the API specified it was the proportion logic high.

Reply:
Ah ok. I think it's simpler to change the polarity when you initialize the timer then.

Reply:
I'm not sure what you mean. Do you want any changes here?

Reply:
This spot should be the same as it is checked in. Change the polarity in the init instead. It is why you are seeing it be low when it is in the active portion.

Reply:
@tannewt oh right the init OCPolarity, facepalm.